src/services/api.tsx

```diff
@@ -19,55 +100,182 @@ const getAge = (birthDate: Date) => {
+      params += "&name=" + x;
+      return params;
+    });
+    let bundlePatient = await makeRequest("Patient", false, params, 10000);
+
+    let entries = bundlePatient.entry;
+    params = "";
+    searchName.split(" ").map((x: string) => {
+      params += "&identifier=" + x;
+      return params;
+    });
+
+    bundlePatient = await makeRequest("Patient", false, params, 10000);
+
+    bundlePatient.entry = bundlePatient.entry.concat(entries);
+
+    bundles.push(bundlePatient);
+  }
+
+  bundles = bundles.concat(
+    await Promise.all(
+      searchParams.map((x: any) => {
+        switch (x.label) {
+          case "Age":
+            params = "";
+            const correspondingDate: Date = new Date();
+            correspondingDate.setFullYear(
+              correspondingDate.getFullYear() - parseInt(x.text)
+            );
+            const yyyy = correspondingDate.getFullYear();
+            const mm =
+              (correspondingDate.getMonth() + 1 > 9 ? "" : "0") +
+              (correspondingDate.getMonth() + 1);
+            const dd =
+              (correspondingDate.getDate() > 9 ? "" : "0") +
+              correspondingDate.getDate();
+
+            switch (x.symbol) {
+              case "=":
+                params += `&birthdate=lt${yyyy}-${mm}-${dd}`;
+                params += `&birthdate=gt${yyyy - 1}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+              case ">":
+                params += `&birthdate=lt${yyyy}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+              case "<":
+                params += `&birthdate=gt${yyyy}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+            }
+            return {};
+          case "Diabète":
+            return getPatientPerCondition("73211009");
+          default:
+            console.info(`Paramètre ${x.label} non reconnu`);
+        }
+        return [];
+      })
+    )
   );
+
+  let finalBundle: Bundle = bundles[0];
+  bundles.map((bundle: Bundle) => {
+    if (finalBundle !== bundle && bundle.entry.length > 0) {
+      let listId = bundle.entry.map((x: any) => x.resource.id);
+      finalBundle.entry = finalBundle.entry.filter(
+        (entry: any) => listId.indexOf(entry.resource.id) >= 0
+      );
+    }
+    return false;
+  });
+  finalBundle.total = finalBundle.entry.length;
+
+  return addPatientsToBundle(finalBundle);
 };
 
-export const getPatientData = async (patientId: string) => {
+export const requestNextPatients = async (bundle: PatientBundle) => {
   /*
-    getPatientData requests for data from Patient resourcce of id patientId
-    return a Patient object.
-  */
-  let response = await client.search({
-    type: "Patient",
-    patient: patientId
+   * function requestNextPatients
+   * Return the same bundle with more patients, fetched from the nextLink attribute.
+   */
+  if (!bundle.nextLink) {
+    console.info("no link available");
```

Comment:
The user will not see the console, maybe you can show him something :)

Reply:
Added in issue #24 
I don't know for now how to manage that for bundle without next link (in case we merge two bundles for example, with complex requests needing multiple api calls)

Reply:
I was just saying that you could show something in the interface instead of printing it to the console (for instance with a toaster as in Pyrog). But if you're going to solve this soon, it's ok